extensions/meme_rater.py

```diff
@@ -0,0 +1,64 @@
+import os
+import hikari, lightbulb
+from PIL import Image
+import requests
+from snark import model
+from google.generativeai.generative_models import GenerativeModel
+from typing import Final
+
+plugin = lightbulb.Plugin("MemeRater")
+
+MEMES_CHANNEL_ID = os.environ.get("MEME_CHANNEL_ID")
+
+# Make this envs if you want
+MEME_RATE_PROMPT: Final[str] = (
+    "Rate this meme out of 10, with 10 being the funniest. Rate is solely on how funny a bunch of computer science nerds would find it. ONLY Return an integer."
+)
+MINIMUM_MEME_RATING_TO_NOT_DELETE: Final[int] = 6  # can be in .env if you prefer
+IMG_FILE_EXTENSIONS: Final = {"jpg", "jpeg", "png", "webp"}
+# removed uncommon formats for testing - don't tell Jimmy plx
+# "tiff",  "bmp", "gif <-- idk if these are supported
+
+
+async def get_meme_rating(image_url: str, model: GenerativeModel):
+    response = model.generate_content(
+        contents=[MEME_RATE_PROMPT, Image.open(requests.get(image_url).raw)]
+    )
+    return response.text
+
+
+@plugin.listener(hikari.GuildMessageCreateEvent)
+async def main(event: hikari.GuildMessageCreateEvent) -> None:
+    if event.channel_id != MEMES_CHANNEL_ID:
+        return
+    for attachment in event.message.attachments:
```

Comment:
I think we should average the rating over all attachments and give one, larger, reply. As it stands, we would reply for each attachment, and possibly give a +1 and a -1, yielding missed messaging.

Reply:
See line 58.

Sent from my phone.

Reply:
See line 58

Reply:
Unless I can't read, I believe that would only abort if the image is bad. Meaning a good meme followed by a bad, is not only a disappointing record, but leads to mixed messaging